ls1intum/Artemis
# [Bugfix] Include only required properties when serializing exerciseUnits

### Checklist

### Motivation and Context


Solves issue #3958 

### Description

I just reproduced the error I found out, that this is only possible for lectures that have exercises as lecture units which contain submissions. During serializing the lecture object for the request body, the cyclic dependency occurs within the nested this nested exercise student participations (results is part of it). I also found out, that the error is not related to setting the start date but rather to any PUT request, i.e. change on the lecture.

Solution: Restrict the `Exercise` properties within a `ExerciseUnit` serialized for a request body to those, that are required. -> We certainly to not need `StudentParticipation` including the results in etc., this led to a cyclic dependency that was reported in the issue #3958 

While fixing the bug, I saw a translation error. "Veröffentlichkeitsdatum" does not exist, I changed it alongside.

<img width="881" alt="Bildschirmfoto 2021-09-28 um 13 13 40" src="https://user-images.githubusercontent.com/41366522/135076586-cc7c0ad8-2832-40fc-8b12-8becb490e38b.png">

### Steps for Testing


1. Log in to Artemis as instructor
2. Navigate to Course Management
3. Find or create a lecture with exercises as lecture units, make sure that these exercises contain some submissions -> on TS1, you can use my lecture https://artemistest.ase.in.tum.de/course-management/477/lectures/184 with user 15
4. Try to edit this lectures, i.e. change title, description, start date, end date
5. After saving, the lecture should be updated, the response for the PUT request should be 200 (Ok)

### Review Progress





- Code Review
- Manual Tests

### Screenshots



Before fixing - only lectures without exercise units can be edited:

https://user-images.githubusercontent.com/41366522/135077792-03942fd0-edea-48c4-9b0e-30a7fbbe8b4f.mov


After fixing - also lectures with exercise units (including submissions) can be edited:

https://user-images.githubusercontent.com/41366522/135076040-00cb2e9c-96e7-48ca-9c99-7fd681d0e6ac.mov
